### ISSUE TITLE
[Hybrid] Allow user to pass in certificate to cwctl when installing Codewind

### ISSUE THREAD

**Issue by johnmcollier** [eclipse/codewind#2438]
Currently when user installs hybrid Codewind, `cwctl` creates a self-signed certificate, stores it in a Kubernetes secret and configures the Codewind gatekeeper and keycloak ingress URLs to use that secret. 

While this is fine for development or trying out Codewind, this makes it hard to set up Codewind with publicly signed certificates.

 It would be nice if a flag was added to cwctl called `--tls-secret` that could be used to reference a secret containing an already existing certificate. This TLS secret would then be used when configuring the Codewind ingresses, rather than creating a new self-signed secret. This would allow users to generate a publicly signed certficate in a secret (e.g. by way of cert manager and let's encrypt) and then pass it into cwctl when installing Codewind

**Comment by tobespc** [comment]
@markcor11 will this be taken care of by the operator work ?

**Comment by markcor11** [comment]
It makes sense to do it in the operator going forward because it will replace cwctl for deploying Codewind.  There are a few places where the self-signed certificates are being generated today including in each of the Codewind containers and it happens when they start up.

Since those certificates will get re-generated on every restart that causes problems for browsers because they would see the certificate as being different each time and warn the user to accept the new certificate.

We can add the ability to accept a TLS cert and key within the Codewind CRD, but we'll also need to make some changes to PFE to use those certificates if they were provided rather than generate new ones each time. 

In follow up steps we could make use of cert-manager and have the certificates generated and signed by the in-cluster CA which will provide an even better experience.

**Comment by johnmcollier** [comment]
@markcor11 I'm good with doing it in the operator, but may not want to store the TLS cert and key in the Codewind CRD. It may be better to accept the Kubernetes TLS secret as an field in the custom resource, like what ingresses do for TLS.

See https://github.com/openshift/origin/issues/2162 for an example of why we wouldn't want to store the tls cert and key directly in the custom resource.

**Comment by markcor11** [comment]
@johnmcollier,  I agree the right place to store the key pair secrets are in Kube secret.   (in fact I'm already doing that for gatekeeper in one called secret-codewind-tls but not for PFE or Perf) 

```
secret-codewind-client-k81235kj          Opaque                                1      12h
secret-codewind-session-k81235kj         Opaque                                1      12h
secret-codewind-tls-k81235kj             Opaque                                2      12h.         <----  key and cert
secret-keycloak-user-k3a237fj            Opaque                                2      5d8h
```

What I mean by adding Ito to the Codewind CRD was to add it to the API but not to persist the keypair there just use it facilitate creating the Kube secret.
